frontend/js/src/user/components/follow/UserSocialNetwork.tsx

```diff
@@ -69,6 +80,82 @@ export default class UserSocialNetwork extends React.Component<
     }
   };
 
+  getSimilarity = async () => {
+    const { user } = this.props;
+    const { APIService, currentUser } = this.context;
+    if (currentUser?.name === user?.name) {
+      return;
+    }
+    const { getSimilarityBetweenUsers } = APIService;
+    try {
+      const response = await getSimilarityBetweenUsers(
+        user.name,
+        currentUser.name
+      );
+      const { payload } = response;
+      const similarityScore = payload.similarity;
+      this.setState({ similarityScore });
+    } catch (err) {
+      if (err.toString() === "Error: Similar-to user not found") {
+        // cannot get similarity if not in the list, so just return 0
+        this.setState({ similarityScore: 0 });
+      } else {
+        toast.error(
+          <ToastMsg
+            title="Error while fetching similarity"
+            message={err.toString()}
+          />,
+          { toastId: "fetch-similarity-error" }
+        );
+      }
+    }
+  };
+
+  getSimilarArtists = async () => {
+    const { user } = this.props;
+    const { APIService, currentUser } = this.context;
+    if (currentUser?.name === user?.name) {
+      return;
+    }
+    const { getUserEntity } = APIService;
+    try {
+      let response = await getUserEntity(
+        user.name,
+        "artist",
+        "all_time",
+        0,
+        100
+      );
+      const userArtists = (response as UserArtistsResponse).payload.artists;
+      response = await getUserEntity(
+        currentUser.name,
+        "artist",
+        "all_time",
+        0,
+        100
+      );
+      const currentUserArtists = (response as UserArtistsResponse).payload
+        .artists;
+      let similarArtists = intersectionBy(
+        userArtists,
+        currentUserArtists,
+        "artist_name"
+      );
+      if (similarArtists.length > 5) {
+        similarArtists = similarArtists.slice(0, 5);
```

Comment:
OK, hear me out... We already have the full list of similar artists, I wonder if we could show only the top 5 (like we do currently) as text, and have the whole list show when we hover over that section of the CompatibilityCard if there are more than 5.
Could be a good way to show more info straight away for those who want, while staying out of the way.
What do you think?

Reply:
i was thinking about this too, but sometimes there are a lot, i have seen upto 40 with a friend of mine. don't know what the best way to display those will be. lemme know if you have any ideas, @Aerozol too

Reply:
so display them as a tool tip when you over the list?